src/Microsoft.AspNetCore.OData/Formatter/ODataInputFormatterFactory.cs

```diff
@@ -72,6 +72,31 @@ private static ODataInputFormatter CreateApplicationJson()
             formatter.SupportedMediaTypes.Add(ODataMediaTypes.ApplicationJsonStreamingTrue);
             formatter.SupportedMediaTypes.Add(ODataMediaTypes.ApplicationJsonStreamingFalse);
             formatter.SupportedMediaTypes.Add(ODataMediaTypes.ApplicationJson);
+            // NOTE: The order in which the media types are added is relevant due to how ASP.NET Core handles content negotiation
```

Comment:
what do you use to order them?

Reply:
@ElizabethOkerio They are ordered in the order they are added to the `SupportedMediaTypes` collection